python/taichi/lang/ast/ast_transformer.py

```diff
@@ -583,7 +591,17 @@ def build_Call(ctx, node):
             node.ptr = func(node.func.caller, *args, **keywords)
             return node.ptr
         ASTTransformer.warn_if_is_external_func(ctx, node)
-        node.ptr = func(*args, **keywords)
+        try:
+            node.ptr = func(*args, **keywords)
+        except TypeError as e:
+            module = inspect.getmodule(func)
+            error_msg = re.sub(r"\bExpr\b", "Taichi Expression", str(e))
+            msg = f"TypeError when calling `{func.__name__}`: {error_msg}."
+            if ASTTransformer.is_external_func(ctx, node.func.ptr):
+                args_has_expr = any([isinstance(arg, Expr) for arg in args])
+                if args_has_expr and (module == math or module == np):
+                    msg += f"\nDid you mean to use `ti.{func.__name__}` instead of `{module.__name__}.{func.__name__}`?"
```

Comment:
```suggestion
                    exec_str = f"from taichi import {func.__name__}"
                    try:
                        exec(exec_str, {})
                    except:
                        pass
                    else:
                        msg += f"\nDid you mean to use `ti.{func.__name__}` instead of `{module.__name__}.{func.__name__}`?"```

Reply:
How about this?

Reply:
this seems dangerous... what if the user did something like `np.ndarray`

Reply:
It just imports function `ndarray` into a temporary global dict (the second argument of `exec`). Why is it dangerous?

Reply:
oh I misunderstood, than this is good I think